### PR TITLE
Add ATLAS into HWIMO-BUILD, Remove duplicated template-packages.json, Update README

### PR DESCRIPTION
**Pull request by panpan0000**
this work is based on request of @changev 

@PengTian0  @RackHD/corecommitters

### CODE REVIEW

File: packer/HWIMO-BUILD [hunk 1]
```diff
@@ -4,7 +4,13 @@
 # input (enviromental variable):
 # * OS_VER        -   ubuntu-14.04 or ubuntu-16.04
 # * BUILD_TYPE    -   vmware or virtualbox
-# * RACKHD_INSTALL_ANSIBLE - which is the ansible yml playbook used. now are rackhd_package or rackhd_local
+# * RACKHD_INSTALL_ANSIBLE - which is the ansible yml playbook used. now are rackhd_ci_builds, rackhd_release or rackhd_local
```
Comment (cgx027) [inline packer/HWIMO-BUILD]: Does the comments means the choice for RACKHD_INSTALL_ANSIBLE can be rackhd_ci_builds, rackhd_release or rackhd_local

File: packer/HWIMO-BUILD [hunk 2]
```diff
@@ -13,18 +19,71 @@
...
+echo '[Info] Please use ENV VAR to pass customized parameter to this script. Please refer to the usage on the top of script file.'
+
+############################  Parameter Process ################ start #############
+
+#Ubuntu Version
 if  [ ! -n "${OS_VER}" ];  then
     OS_VER=ubuntu-14.04
 fi
-echo "Packer Build based on $OS_VER"
+echo "[Info] Packer Build based on $OS_VER"
```
Comment (cgx027) [inline packer/HWIMO-BUILD]: consider set [Info] as a variable and use ```echo $info_header Packer Build based on $OS_VER```

File: packer/HWIMO-BUILD [hunk 3]
```diff
@@ -4,7 +4,13 @@
 # input (enviromental variable):
 # * OS_VER        -   ubuntu-14.04 or ubuntu-16.04
 # * BUILD_TYPE    -   vmware or virtualbox
```
Comment (cgx027) [inline packer/HWIMO-BUILD]: maybe add some info there stating the default value is virtualbox, as the code indicts.

File: packer/HWIMO-BUILD [hunk 4]
```diff
@@ -4,7 +4,13 @@
...
-# * RACKHD_INSTALL_ANSIBLE - which is the ansible yml playbook used. now are rackhd_package or rackhd_local
+# * RACKHD_INSTALL_ANSIBLE - which is the ansible yml playbook used. now are rackhd_ci_builds, rackhd_release or rackhd_local
+# * RACKHD_VERSION        - optional , by default is blank. used for rackhd_ci_builds.yml to install specific version of rackhd.deb, and moreover, this version used to version the vagrant box when uploading to ATLAS
+# * UPLOAD_BOX_TO_ATLAS   - optional, by default is False. tell packer if upload the vagrant box generated to ATLAS. NOTE: ATLAS user/token and target ATLAS repo name should be specificed.
```
Comment (cgx027) [inline packer/HWIMO-BUILD]: The code seems expecting ```true/false``` or ```TRUE/FALSE```, suggest change from ```False``` to ```false``` or ```FALSE```.

File: packer/HWIMO-BUILD [hunk 5]
```diff
@@ -4,7 +4,13 @@
 # input (enviromental variable):
 # * OS_VER        -   ubuntu-14.04 or ubuntu-16.04
 # * BUILD_TYPE    -   vmware or virtualbox
-# * RACKHD_INSTALL_ANSIBLE - which is the ansible yml playbook used. now are rackhd_package or rackhd_local
+# * RACKHD_INSTALL_ANSIBLE - which is the ansible yml playbook used. now are rackhd_ci_builds, rackhd_release or rackhd_local
+# * RACKHD_VERSION        - optional , by default is blank. used for rackhd_ci_builds.yml to install specific version of rackhd.deb, and moreover, this version used to version the vagrant box when uploading to ATLAS
```
Comment (cgx027) [inline packer/HWIMO-BUILD]: Any RACK_VERSION format convention needs to be specified here?

File: packer/HWIMO-BUILD [hunk 6]
```diff
@@ -13,18 +19,71 @@
...
+# By default. use ansible playbook : rackhd_local.yml
+# can become an ENV VAR in the future
+if  [ ! -n "${RACKHD_INSTALL_ANSIBLE}" ];  then
+    RACKHD_INSTALL_ANSIBLE=rackhd_local
+fi
+echo "[Info] Install RackHD with ansible :  ${RACKHD_INSTALL_ANSIBLE}.yml "
+
 
 
+if [  -n "${RACKHD_VERSION}" ];  then
```
Comment (cgx027) [inline packer/HWIMO-BUILD]: I am not sure if there is a need to check RACKHD_VERSION against a regex. Just bring it up.

File: packer/HWIMO-BUILD [hunk 7]
```diff
@@ -13,18 +19,71 @@
...
+           echo "[Error]: ATLAS_VERSION(${ATLAS_VERSION}) should be X.Y.Z (X.Y.Z is digit), that's only format accepted by ATLAS."
+          exit 2
+      fi
+   fi
+fi
+
+
+# By default. use ansible playbook : rackhd_local.yml
+# can become an ENV VAR in the future
+if  [ ! -n "${RACKHD_INSTALL_ANSIBLE}" ];  then
```
Comment (PengTian0) [inline packer/HWIMO-BUILD]: Is it necessary to check "RACKHD_INSTALL_ANSIBLE" in [rackhd_ci_builds,  rackhd_release,  rackhd_local] ?

Or check whether the script $RACKHD_INSTALL_ANSIBLE.yml exists?

**Comment by panpan0000** [comment]
test this please

File: packer/HWIMO-BUILD (hunk 2 above)
Comment (panpan0000) [inline packer/HWIMO-BUILD]: done, thx

File: packer/HWIMO-BUILD (hunk 3 above)
Comment (panpan0000) [inline packer/HWIMO-BUILD]: done, thx

File: packer/HWIMO-BUILD (hunk 4 above)
Comment (panpan0000) [inline packer/HWIMO-BUILD]: done, thx

File: packer/HWIMO-BUILD [hunk 8]
```diff
@@ -48,9 +100,22 @@ VMDIR=output-${BUILD_TYPE}-iso
...
-echo { \"playbook\": \"${RACKHD_INSTALL_ANSIBLE}\",   > template.cfg
-echo  \"vm_name\": \"${VM_NAME}\" } >> template.cfg
+echo {                                                > $CFG_FILE
+echo  \"playbook\": \"${RACKHD_INSTALL_ANSIBLE}\",   >> $CFG_FILE
+echo  \"vm_name\": \"${VM_NAME}\"                    >> $CFG_FILE
+
+if [[ $IS_UPLOAD_ATLAS == true ]]
+then
+    echo ,                                           >> $CFG_FILE
+    echo \"atlas_username\": \"${ATLAS_USERNAME}\" , >> $CFG_FILE
```
Comment (panpan0000) [inline packer/HWIMO-BUILD]: Good Catch !!!  change to ENV Variable.
@changev , F.Y.I.

File: packer/HWIMO-BUILD (hunk 5 above)
Comment (panpan0000) [inline packer/HWIMO-BUILD]: Nope... It's  not decided yeah..
but I added comments to refer to bintray.com/rackhd to find correct version..

File: packer/HWIMO-BUILD (hunk 6 above)
Comment (panpan0000) [inline packer/HWIMO-BUILD]: same as above. the version format is still on fly...

File: packer/HWIMO-BUILD (hunk 1 above)
Comment (panpan0000) [inline packer/HWIMO-BUILD]: done, thx

File: packer/HWIMO-BUILD (hunk 5 above)
Comment (panpan0000) [inline packer/HWIMO-BUILD]: fixed...

File: packer/HWIMO-BUILD [hunk 9]
```diff
@@ -13,18 +19,71 @@
...
+fi
+
+echo "[Info] Packer Build : ${BUILD_TYPE}"
+
+
+IS_UPLOAD_ATLAS=false
+if [[ "${UPLOAD_BOX_TO_ATLAS}" == "true" ]] ||  [[ "${UPLOAD_BOX_TO_ATLAS}" == "TRUE" ]];
+then
+    IS_UPLOAD_ATLAS=true
+fi
```
Comment (panpan0000) [inline packer/HWIMO-BUILD]: Nice . I like it !

File: packer/HWIMO-BUILD (hunk 7 above)
Comment (panpan0000) [inline packer/HWIMO-BUILD]: done, add "-f" to check file exist or not.

**Comment by panpan0000** [comment]
Hi, @changev  @cgx027 @PengTian0 , 
I add a new commit to address your comments :  https://github.com/RackHD/RackHD/pull/497/commits/a9ff37af315e7636253199e8ea11a019ad52e8b4

**Comment by panpan0000** [comment]
Hi, @RackHD/corecommitters , Can you approve this PR?

**Comment by anhou** [comment]
@panpan0000 please resolve the conflicts

**Comment by panpan0000** [comment]
conflicts resolved. Thanks.

**Pull request merged**